### ISSUE TITLE
How does this package compare to flyspell?

### ISSUE THREAD

**Issue by ghost** [leotaku/flycheck-aspell#12]
Stock Emacs already has a package called "flyspell-mode". It is rather old and sometimes behaves weirdly, but I think it would still be nice to have a small introduction at the README, how flycheck-aspell is better.

Could that be added?

**Comment by leotaku** [comment]
@lockywolf I think this is a good idea. I also think the whole README deserves a rework, thanks for reminding me of that.

What do you think about the following comparison? I personally think it's a bit wordy.

---

#### Differences to `flyspell-mode`

Emacs already contains a mode for spell checking called `flyspell-mode`. It works by spell-checking words as you type, which allows for constant performance with arbitrarily large files, but also comes with a number of disadvantages.  In particular, it cannot highlight misspelled words that have not been near the cursor in your current editing session. Accessing a list of all misspelled words is also impossible. Lastly, because `flyspell-mode` has to run code on every keystroke, it can often cause noticeable input delay and stuttering. 

In contrast, `flycheck-aspell` and `flymake-aspell` asynchronously run Aspell in the background and extract a list of all misspelled words in the full document. This ensures a list of misspelled words is always available, which can be accessed with e.g. `flymake-show-diagnostics-buffer` or `flycheck-list-errors`. However, this also means that spell checking performance will slow down approximately linearly with the number of spelling errors. If your file is particularly large it might take a few seconds for new misspelled words to be highlighted. Note that this should never affect editing experience as no additional code is run on user input.

---

**Comment by leotaku** [comment]
I have just rewritten a large part of the README, incorporating your suggestion of adding a comparison to `flyspell-mode`. Feel free to leave any further suggestions here or open a new issue or discussion.
